MapboxCoreNavigation/EventDetails.swift

```diff
@@ -5,8 +5,44 @@ import UIKit
 import AVFoundation
 import MapboxDirections
 
+protocol EventDetails: Encodable {
+    var event: String? { get }
+    var created: Date { get }
+    var sessionIdentifier: String { get }
+}
+
+struct PerformanceEventDetails: EventDetails {
+    let event: String?
+    let created: Date
+    let sessionIdentifier: String
+    var counters: [String: Double] = [:]
+    var attributes: [String: String] = [:]
```

Comment:
We made a late-breaking decision to store counters and attributes as flat dictionaries instead of as arrays of key-value dictionaries.

Reply:
Never mind, we decided against storing the counters and attributes as flat dictionaries.